### PR TITLE
removed unused broadcast metric

### PR DESCRIPTION
**Pull request by pompon0**
The code was used in integration tests which has since been deleted.
Additionally narrowed down visibility of network metrics.

### CODE REVIEW

File: chain/network/src/stats/metrics.rs [hunk 1]
```diff
@@ -1,122 +1,15 @@
-use crate::network_protocol::Encoding;
-use near_metrics::{
-    exponential_buckets, try_create_histogram, try_create_histogram_vec, try_create_int_counter,
-    try_create_int_counter_vec, try_create_int_gauge, Histogram, HistogramVec, IntCounter,
-    IntCounterVec, IntGauge, IntGaugeVec,
-};
-use near_network_primitives::types::{PeerType, RoutedMessageBody};
+use near_network_primitives::types::RoutedMessageBody;
```
Comment (akhi3030) [inline chain/network/src/stats/metrics.rs:1]: I am trying to understand why there are still some metrics left in this file.  Can you explain it please?

File: chain/network/src/test_utils.rs [hunk 2]
```diff
@@ -219,23 +219,6 @@ impl Handler<GetInfo> for PeerManagerActor {
     }
 }
 
-/// `GetBroadcastMessageCount` gets `NetworkMetrics` from `PeerManager`.
```
Comment (akhi3030) [inline chain/network/src/test_utils.rs:222]: Always happy to see deadcode being removed.  Would it be better to split unrelated changes into separate PRs?

File: chain/network/src/stats/metrics.rs (hunk 1 above)
Comment (pompon0) [reply chain/network/src/stats/metrics.rs:1]: The remaining ones are updated in multiple locations (it's not really a best practice), and fixing that has to wait for a refactor.

File: chain/network/src/test_utils.rs (hunk 2 above)
Comment (pompon0) [reply chain/network/src/test_utils.rs:222]: The removed deadcode is a getter of an unused metric, so it is related.

**Comment by matklad** [comment]
I am a bit concerned that this breaks consistency: today, *all* metrcs are created in various `metrics.rs` files, so theres' only one place to look at for metrics, irrespective of the component you work with. 

After this PR, the network would be an odd one out. 

My gut feeling is that we should keep the net consistent with other stuff, even if, abstractly, collecting all component's metrics into a single `metrics.rs` file is worse.

**Comment by akhi3030** [comment]
Thanks for your perspective matklad!  This makes sense to me as well.

**Comment by pompon0** [comment]
@matklad how about multiple embedded "metric" modules? Would that be consistent enough?

**Comment by mina86** [comment]
Just one more thing I’d like to add:

> AFAIC even the name of the metric should include a path to the code location which manages a given metric object.

I don’t think that’s a good idea.  This would mean name of the metric would change each time we refactor code and move things around.  Metrics are essentially an API and should not needlessly expose implementation details.

**Comment by akhi3030** [comment]
I wonder if in future it makes sense to write a short RFC doc about such proposals to gather feedback.

**Comment by pompon0** [comment]
ok, narrowed the PR scope to just remove the deadcode

**Review by mina86 — APPROVED** [review]

File: chain/network/src/stats/metrics.rs [hunk 3]
```diff
@@ -83,26 +81,26 @@ pub static REQUEST_COUNT_BY_TYPE_TOTAL: Lazy<IntCounterVec> = Lazy::new(|| {
 });
 
 // Routing table metrics
-pub static ROUTING_TABLE_RECALCULATIONS: Lazy<IntCounter> = Lazy::new(|| {
+pub(crate) static ROUTING_TABLE_RECALCULATIONS: Lazy<IntCounter> = Lazy::new(|| {
```
Comment (akhi3030) [inline chain/network/src/stats/metrics.rs:84]: Can you highlight this change in the description of the PR please?

**Review by akhi3030 — APPROVED** [review]

File: chain/network/src/stats/metrics.rs (hunk 3 above)
Comment (pompon0) [reply chain/network/src/stats/metrics.rs:84]: done
